src/Commands/Connection/SetCommand.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Connection;
+
+use Consolidation\OutputFormatters\StructuredData\RowsOfFields;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Terminus\Collections\Sites;
+use Terminus\Exceptions\TerminusException;
+use Terminus\Models\Environment;
+
+/**
+ * Class SetCommand
+ * @package Pantheon\Terminus\Commands\Connection
+ */
+class SetCommand extends TerminusCommand
+{
+    /**
+     * Set git or sftp connection mode on a site's dev or multidev environment
+     *
+     * @authorized
+     *
+     * @command connection:set
+     *
+     * @param string $environment Name of the environment to set. Note that you cannot change 'test' or 'live'.
+     * @param string $mode        Connection mode, one of: <git|sftp>
+     *
+     * @return bool
+     *
+     */
+    public function connectionSet($environment, $mode)
+    {
+        $site_env = explode('.', $environment);
+        $sites = new Sites();
+        $site = $sites->get($site_env[0]);
+        $env   = $site->environments->get($site_env[1]);
+
+        $workflow = $env->changeConnectionMode($mode);
+        if (is_string($workflow)) {
+            $this->log()->notice($workflow);
+        } else {
+            while (!$workflow->checkProgress()) {
+                // TODO: (ajbarry) Add workflow progress output
+            }
+            $this->log()->notice($workflow->getMessage());
```

Comment:
@marktheunissen thanks for the review. @TeslaDethray has some other ideas for the utility in `workflowOutput` riffing off of what @ronan introduced which is the pattern followed here.